libnest/src/package/mod.rs

```diff
@@ -54,72 +60,64 @@ impl Default for Kind {
 #[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default, Serialize, Deserialize)]
 pub struct PackageName(String);
 
-impl PackageName {
-    /// Create a [`PackageName`] from a String
-    pub fn from(name: String) -> Self {
-        PackageName(name)
-    }
-}
+macro_rules! strong_name_impl {
+    ($NameType:ident, $RegexValidator:ident) => {
```

Comment:
Ideally `$RegexValidator` should probably be an expression, it would allow in place creation of Regex instead of giving only an indentifier pointing to them. It's not a real issue though, this macro is only local to the current module, but for next time, would be neat =)